langchain_tools_demo/agent.py

```diff
@@ -21,20 +21,23 @@
 from langchain.memory import ConversationBufferMemory
 from langchain.prompts.chat import ChatPromptTemplate
 
-from tools import convert_date, tools
+from tools import convert_date, initialize_tools
+import aiohttp
 
 set_verbose(bool(os.getenv("DEBUG", default=False)))
+session = None
 
 
 # Agent
-def init_agent() -> AgentExecutor:
+async def init_agent() -> AgentExecutor:
```

Comment:
This should probably return AgentExecutor and ClientSession in a new object so we retain references to them both

Reply:
created new class ClientAgent